File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>m;

        vector<int>output;
        for(int i=0;i<nums.size();i++)
        {
            int complement=target-nums[i];
            if(m.find(complement)!=m.end()){
                output.push_back(i);
                output.push_back(m[complement]);
                return output;
            }
         
            m[nums[i]]=i;
        }

        return output;
        
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums, target):
        seen = {}
        
        for i, num in enumerate(nums):
            complement = target - num
            if complement in seen:
                return [seen[complement], i]  # Return index of complement and current num
            seen[num] = i  # Store index of current num
        
        return []